### ISSUE TITLE
AdversarialFairnessClassifier has no attribute 'classes__'

### ISSUE THREAD

**Issue by haytham918** [fairlearn/fairlearn#1381]
So I was running the piece of code from #1373, I just added `scoring="accuracy"` or something else like precision, f1, roc_auc for the grid search, it gives me the error that it has no attribute "classes__"

Code:
```
from sklearn.compose import make_column_transformer, make_column_selector
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.impute import SimpleImputer
from sklearn.model_selection import train_test_split
from sklearn.pipeline import Pipeline
from numpy import number
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import GridSearchCV
from fairlearn.adversarial import AdversarialFairnessClassifier
from fairlearn.datasets import fetch_adult

import sklearn

sklearn.set_config(enable_metadata_routing=True)

X, y = fetch_adult(return_X_y=True)
pos_label = y[0]

z = X["sex"]  # In this example, we consider 'sex' the sensitive feature.

ct = make_column_transformer(
    (
        Pipeline(
            [
                ("imputer", SimpleImputer(strategy="mean")),
                ("normalizer", StandardScaler()),
            ]
        ),
        make_column_selector(dtype_include=number),
    ),
    (
        Pipeline(
            [
                ("imputer", SimpleImputer(strategy="most_frequent")),
                ("encoder", OneHotEncoder(drop="if_binary", sparse_output=False)),
            ]
        ),
        make_column_selector(dtype_include="category"),
    ),
)

X_train, X_test, Y_train, Y_test, Z_train, Z_test = train_test_split(
    X, y, z, test_size=0.2, random_state=12345, stratify=y
)

X_prep_train = ct.fit_transform(X_train)  # Only fit on training data!
X_prep_test = ct.transform(X_test)

mitigator = AdversarialFairnessClassifier(
    backend="torch",
    predictor_model=[50, "leaky_relu"],
    adversary_model=[3, "leaky_relu"],
    batch_size=2**8,
    progress_updates=0.5,
    random_state=123,
).set_fit_request(sensitive_features=True)

pipe = make_pipeline(ct, mitigator)

param_grid = {
    "adversarialfairnessclassifier__batch_size": [2**8, 2**6],
}

gs = GridSearchCV(pipe, param_grid, cv=5, scoring="accuracy").fit(
    X_train, Y_train, sensitive_features=Z_train
)
```

Error:
```
warnings.warn(
/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/model_selection/_validation.py:993: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/model_selection/_validation.py", line 982, in _score
    scores = scorer(estimator, X_test, y_test, **score_params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/metrics/_scorer.py", line 253, in __call__
    return self._score(partial(_cached_call, None), estimator, X, y_true, **_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/metrics/_scorer.py", line 345, in _score
    y_pred = method_caller(
             ^^^^^^^^^^^^^^
  File "/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/metrics/_scorer.py", line 87, in _cached_call
    result, _ = _get_response_values(
                ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/utils/_response.py", line 198, in _get_response_values
    classes = estimator.classes_
              ^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/haytham/lib/python3.12/site-packages/sklearn/pipeline.py", line 1012, in classes_
    return self.steps[-1][1].classes_
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'AdversarialFairnessClassifier' object has no attribute 'classes_'
```

**Comment by adrinjalali** [comment]
Thanks for the report. We'll fix this soon.

**Comment by TamaraAtanasoska** [comment]
as this was fixed in #1384, we can now close the issue (cc @adrinjalali)
